src/textual/dom.py

```diff
@@ -278,7 +278,7 @@ def screen(self) -> "Screen":
         while node and not isinstance(node, Screen):
             node = node._parent
         if not isinstance(node, Screen):
-            raise NoScreen(f"{self} has no screen")
+            raise NoScreen(f"node has no screen")
```

Comment:
Perhaps wanted `node` inside a pair of `{}` here?

Reply:
No, the problem was that the repr in the exception message could refer to `self.screen` which would create a infinite loop!

Reply:
Sorry, perhaps I wasn't clear. In the replacement line you're using an f-string with no variable content; so perhaps the intention was to have `f"{node} has no screen"` rather than `f"node has no screen"`.

Reply:
Ahh, I see; was being thrown by the intention of it being an f-string with nothing to f.